gerber/lib/Gerber/Attribute/FileFunction/Drill.hs

```diff
@@ -0,0 +1,37 @@
+{-# language OverloadedStrings #-}
+
+module Gerber.Attribute.FileFunction.Drill
+  ( Type(..), parseType
+  , Via(..), parseVia
+  ) where
+
+-- gerber
+import Gerber.Attribute.Attribute ( Field )
+
+-- text
+import Data.Text ( unpack )
+
+
+data Type = Drill | Route | Mixed
+  deriving ( Eq, Show )
+
+
+parseType :: MonadFail m => Field -> m Type
+parseType field = case field of
+  "Drill" -> pure Drill
+  "Route" -> pure Route
+  "Mixed" -> pure Mixed
+  _ -> fail $ "Bad Drill.Type: " <> unpack field
+
+
+data Via = TH | Blind | Buried
+  deriving ( Eq, Show )
+
+
+parseVia :: MonadFail m => Field -> m Via
+parseVia field = case field of
+  "PTH" -> pure TH
+  "NPTH" -> pure TH
```

Comment:
why don't we have a distinct constructor for non plated through hole, I feel we are loosing vital information here, also you won't be able to print what you parse

Reply:
I drop that information because it's redundant, and it allows me to use the same record for both the fields of both `Plated` and `NonPlated`. Here's the relevant part of the ![the spec](https://user-images.githubusercontent.com/59563355/105489108-9f324680-5caa-11eb-9dc3-e59ff5d41fed.png).

Reply:
:+1: